src/node.rs

```diff
@@ -15,8 +15,22 @@ pub trait NodeExt<'a> {
 
 impl<'a> NodeExt<'a> for Node<'a> {
     fn is_descendant_of(&self, node: Node) -> bool {
-        self.range().start_byte >= node.range().start_byte
-            && self.range().end_byte <= node.range().end_byte
+        if self.start_byte() < node.start_byte() {
+            return false;
+        }
+        if self.end_byte() > node.end_byte() {
+            return false;
+        }
+        if self.start_byte() == node.start_byte() && self.end_byte() == node.end_byte() {
+            let mut ancestor = return_default_if_none!(self.parent());
+            loop {
```

Comment:
Could probably technically do this more efficiently by checking that byte ranges are still identical as we're looping through ancestors (ie we could bail as soon as we see an ancestor with a different byte range)